[Bug] Build fails if no cluster running

**Describe the bug**

Running `make all` fails during the tests if there is no kubernetes cluster running.

**To Reproduce**
Steps to reproduce the behavior:
1. clone master
2. `make all`

**Expected behavior**
If possible warn that tests failed and continue build

**Environment (please complete the following information):**
 - Strimzi version: master
 - Installation method: n/a
 - Kubernetes cluster: none

**YAML files and logs**
```
[INFO] Running io.strimzi.api.kafka.model.KafkaUserCrdIT
ERROR StatusLogger No Log4j 2 configuration file found. Using default configuration (logging only errors to the console), or user programmatically provided configurations. Set system property 'log4j2.debug' to show Log4j 2 internal initialization logging. See https://logging.apache.org/log4j/2.x/manual/configuration.html for instructions on how to configure Log4j 2
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 53.869 s <<< FAILURE! - in io.strimzi.api.kafka.model.KafkaUserCrdIT
[ERROR] io.strimzi.api.kafka.model.KafkaUserCrdIT  Time elapsed: 53.868 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.junit.platform.commons.util.ReflectionUtils.newInstance(ReflectionUtils.java:453)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:62)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.invokeTestClassConstructor(ClassTestDescriptor.java:345)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:290)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:281)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateAndPostProcessTestInstance(ClassTestDescriptor.java:269)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$testInstancesProvider$2(ClassTestDescriptor.java:259)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$testInstancesProvider$3(ClassTestDescriptor.java:263)
	at java.util.Optional.orElseGet(Optional.java:267)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$testInstancesProvider$4(ClassTestDescriptor.java:262)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$before$0(ClassTestDescriptor.java:192)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.before(ClassTestDescriptor.java:191)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.before(ClassTestDescriptor.java:77)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:229)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:197)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:191)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:117)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.RuntimeException: Unable to find a cluster; tried [oc, minikube, minishift]
	at io.strimzi.test.k8s.KubeCluster.bootstrap(KubeCluster.java:82)
	at io.strimzi.test.k8s.KubeClusterResource.cluster(KubeClusterResource.java:79)
	at io.strimzi.test.k8s.KubeClusterResource.cmdClient(KubeClusterResource.java:58)
	at io.strimzi.test.k8s.KubeClusterResource.defaultNamespace(KubeClusterResource.java:53)
	at io.strimzi.test.BaseITST.<clinit>(BaseITST.java:38)
	... 50 more
```

--- a/api/src/test/java/io/strimzi/api/kafka/model/AbstractCrdIT.java
+++ b/api/src/test/java/io/strimzi/api/kafka/model/AbstractCrdIT.java
@@ -41,13 +41,13 @@ public abstract class AbstractCrdIT extends BaseITST {
         RuntimeException thrown2 = null;
         try {
             try {
-                CLUSTER.cmdClient().applyContent(ssStr);
+                kubeCluster().cmdClient().applyContent(ssStr);
             } catch (RuntimeException t) {
                 thrown = t;
             }
         } finally {
             try {
-                CLUSTER.cmdClient().deleteContent(ssStr);
+                kubeCluster().cmdClient().deleteContent(ssStr);
             } catch (RuntimeException t) {
                 thrown2 = t;
             }
@@ -64,6 +64,6 @@ public abstract class AbstractCrdIT extends BaseITST {
 
     @Before
     public void setupTests() {
-        CLUSTER.before();
+        kubeCluster().before();
     }
 }

--- a/cluster-operator/src/test/java/io/strimzi/operator/cluster/MainIT.java
+++ b/cluster-operator/src/test/java/io/strimzi/operator/cluster/MainIT.java
@@ -7,11 +7,14 @@ package io.strimzi.operator.cluster;
 import io.fabric8.kubernetes.client.DefaultKubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.strimzi.operator.common.operator.resource.ClusterRoleOperator;
+import io.strimzi.test.k8s.KubeCluster;
+import io.strimzi.test.k8s.NoClusterException;
 import io.vertx.core.Vertx;
 import io.vertx.ext.unit.Async;
 import io.vertx.ext.unit.TestContext;
 import io.vertx.ext.unit.junit.VertxUnitRunner;
 import org.junit.After;
+import org.junit.Assume;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -38,6 +41,11 @@ public class MainIT {
 
     @Test
     public void testCreateClusterRoles(TestContext context) {
+        try {
+            KubeCluster.bootstrap();
+        } catch (NoClusterException e) {
+            Assume.assumeTrue(e.getMessage(), false);
+        }
         Map<String, String> envVars = new HashMap<>(1);
         envVars.put(ClusterOperatorConfig.STRIMZI_CREATE_CLUSTER_ROLES, "TRUE");
         ClusterOperatorConfig config = ClusterOperatorConfig.fromMap(envVars);

--- a/kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java
+++ b/kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java
@@ -144,8 +144,15 @@ public class KafkaAgent {
     }
 
     private void touch(File file) throws IOException {
-        new FileOutputStream(file).close();
-        file.deleteOnExit();
+        FileOutputStream out = null;
+        try {
+            out = new FileOutputStream(file);
+            file.deleteOnExit();
+        } finally {
+            if (out != null) {
+                out.close();
+            }
+        }
     }
 
     /**

--- a/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorIT.java
+++ b/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorIT.java
@@ -9,12 +9,15 @@ import io.fabric8.kubernetes.api.model.KubernetesResourceList;
 import io.fabric8.kubernetes.client.DefaultKubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.dsl.Resource;
+import io.strimzi.test.k8s.KubeCluster;
+import io.strimzi.test.k8s.NoClusterException;
 import io.vertx.core.Future;
 import io.vertx.core.Vertx;
 import io.vertx.ext.unit.Async;
 import io.vertx.ext.unit.TestContext;
 import io.vertx.ext.unit.junit.VertxUnitRunner;
 import org.junit.AfterClass;
+import org.junit.Assume;
 import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -33,13 +36,20 @@ public abstract class AbstractNonNamespacedResourceOperatorIT<C extends Kubernet
 
     @BeforeClass
     public static void before() {
+        try {
+            KubeCluster.bootstrap();
+        } catch (NoClusterException e) {
+            Assume.assumeTrue(e.getMessage(), false);
+        }
         vertx = Vertx.vertx();
         client = new DefaultKubernetesClient();
     }
 
     @AfterClass
     public static void after() {
-        vertx.close();
+        if (vertx != null) {
+            vertx.close();
+        }
     }
 
     abstract AbstractNonNamespacedResourceOperator operator();

--- a/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperatorIT.java
+++ b/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperatorIT.java
@@ -11,12 +11,15 @@ import io.fabric8.kubernetes.api.model.NamespaceBuilder;
 import io.fabric8.kubernetes.client.DefaultKubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.dsl.Resource;
+import io.strimzi.test.k8s.KubeCluster;
+import io.strimzi.test.k8s.NoClusterException;
 import io.vertx.core.Future;
 import io.vertx.core.Vertx;
 import io.vertx.ext.unit.Async;
 import io.vertx.ext.unit.TestContext;
 import io.vertx.ext.unit.junit.VertxUnitRunner;
 import org.junit.AfterClass;
+import org.junit.Assume;
 import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -37,6 +40,11 @@ public abstract class AbstractResourceOperatorIT<C extends KubernetesClient, T e
 
     @BeforeClass
     public static void before() {
+        try {
+            KubeCluster.bootstrap();
+        } catch (NoClusterException e) {
+            Assume.assumeTrue(e.getMessage(), false);
+        }
         vertx = Vertx.vertx();
         client = new DefaultKubernetesClient();
 
@@ -58,7 +66,9 @@ public abstract class AbstractResourceOperatorIT<C extends KubernetesClient, T e
 
     @AfterClass
     public static void after() {
-        vertx.close();
+        if (vertx != null) {
+            vertx.close();
+        }
     }
 
     abstract AbstractResourceOperator operator();

--- a/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaCrdOperatorIT.java
+++ b/operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaCrdOperatorIT.java
@@ -14,8 +14,10 @@ import io.strimzi.api.kafka.model.DoneableKafka;
 import io.strimzi.api.kafka.model.Kafka;
 import io.strimzi.api.kafka.model.KafkaBuilder;
 import io.strimzi.api.kafka.model.status.ConditionBuilder;
-import io.strimzi.operator.PlatformFeaturesAvailability;
 import io.strimzi.operator.KubernetesVersion;
+import io.strimzi.operator.PlatformFeaturesAvailability;
+import io.strimzi.test.k8s.KubeCluster;
+import io.strimzi.test.k8s.NoClusterException;
 import io.vertx.core.Future;
 import io.vertx.core.Vertx;
 import io.vertx.ext.unit.Async;
@@ -24,6 +26,7 @@ import io.vertx.ext.unit.junit.VertxUnitRunner;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.junit.AfterClass;
+import org.junit.Assume;
 import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -47,6 +50,11 @@ public class KafkaCrdOperatorIT {
 
     @BeforeClass
     public static void before() {
+        try {
+            KubeCluster.bootstrap();
+        } catch (NoClusterException e) {
+            Assume.assumeTrue(e.getMessage(), false);
+        }
         vertx = Vertx.vertx();
         client = new DefaultKubernetesClient();
         kafkaOperator = new CrdOperator(vertx, client, Kafka.class, KafkaList.class, DoneableKafka.class);
@@ -74,10 +82,14 @@ public class KafkaCrdOperatorIT {
 
     @AfterClass
     public static void after() {
-        log.info("Deleting CRD");
-        client.customResourceDefinitions().delete(Crds.kafka());
+        if (client != null) {
+            log.info("Deleting CRD");
+            client.customResourceDefinitions().delete(Crds.kafka());
+        }
 
-        vertx.close();
+        if (vertx != null) {
+            vertx.close();
+        }
     }
 
     protected Kafka getResource() {

--- a/systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java
+++ b/systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java
@@ -117,7 +117,7 @@ public abstract class AbstractST extends BaseITST implements TestSeparator {
     Random rng = new Random();
 
     protected HelmClient helmClient() {
-        return CLUSTER.helmClient().namespace(getNamespace());
+        return kubeCluster().helmClient().namespace(getNamespace());
     }
 
     static String kafkaClusterName(String clusterName) {

--- a/test/src/main/java/io/strimzi/test/BaseITST.java
+++ b/test/src/main/java/io/strimzi/test/BaseITST.java
@@ -41,7 +41,7 @@ public class BaseITST {
 
     public static final Config CONFIG = Config.autoConfigure(System.getenv().getOrDefault("TEST_CLUSTER_CONTEXT", null));
 
-    public static KubeClusterResource kubeCluster() {
+    public static synchronized KubeClusterResource kubeCluster() {
         if (cluster == null) {
             try {
                 cluster = KubeClusterResource.getKubeClusterResource();

--- a/test/src/main/java/io/strimzi/test/BaseITST.java
+++ b/test/src/main/java/io/strimzi/test/BaseITST.java
@@ -12,6 +12,7 @@ import io.strimzi.test.timemeasuring.Operation;
 import io.strimzi.test.timemeasuring.TimeMeasuringSystem;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Assumptions;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.TestInfo;
@@ -34,10 +35,24 @@ public class BaseITST {
     private static final Logger LOGGER = LogManager.getLogger(BaseITST.class);
     protected static final String CLUSTER_NAME = "my-cluster";
 
-    public static final KubeClusterResource CLUSTER = KubeClusterResource.getKubeClusterResource();
-    private static String namespace = CLUSTER.defaultNamespace();
+    private static KubeClusterResource cluster;
+    private static String namespace;
+    private static String defaultNamespace;
 
     public static final Config CONFIG = Config.autoConfigure(System.getenv().getOrDefault("TEST_CLUSTER_CONTEXT", null));
+
+    public static KubeClusterResource kubeCluster() {
+        if (cluster == null) {
+            try {
+                cluster = KubeClusterResource.getKubeClusterResource();
+                namespace = cluster.defaultNamespace();
+                defaultNamespace = cluster.cmdClient().defaultNamespace();
+            } catch (RuntimeException e) {
+                Assumptions.assumeTrue(false, e.getMessage());
+            }
+        }
+        return cluster;
+    }
 
     /**
      * Sets the namespace value for Kubernetes clients
@@ -63,7 +78,7 @@ public class BaseITST {
      * @return CMD client
      */
     public static KubeCmdClient<?> cmdKubeClient() {
-        return CLUSTER.cmdClient().namespace(namespace);
+        return kubeCluster().cmdClient().namespace(namespace);
     }
 
     /**
@@ -72,7 +87,7 @@ public class BaseITST {
      * @return CMD client with expected namespace in configuration
      */
     public static KubeCmdClient<?> cmdKubeClient(String inNamespace) {
-        return CLUSTER.cmdClient().namespace(inNamespace);
+        return kubeCluster().cmdClient().namespace(inNamespace);
     }
 
     /**
@@ -80,7 +95,7 @@ public class BaseITST {
      * @return Kubernetes client
      */
     public static KubeClient kubeClient() {
-        return CLUSTER.client().namespace(namespace);
+        return kubeCluster().client().namespace(namespace);
     }
 
     /**
@@ -89,12 +104,10 @@ public class BaseITST {
      * @return Kubernetes client with expected namespace in configuration
      */
     public static KubeClient kubeClient(String inNamespace) {
-        return CLUSTER.client().namespace(inNamespace);
+        return kubeCluster().client().namespace(inNamespace);
     }
 
-    private static final String DEFAULT_NAMESPACE = CLUSTER.cmdClient().defaultNamespace();
-
-    protected String clusterOperatorNamespace = DEFAULT_NAMESPACE;
+    protected String clusterOperatorNamespace = defaultNamespace;
     protected List<String> bindingsNamespaces = new ArrayList<>();
     public List<String> deploymentNamespaces = new ArrayList<>();
     private List<String> deploymentResources = new ArrayList<>();
@@ -179,8 +192,8 @@ public class BaseITST {
             cmdKubeClient().waitForResourceDeletion("Namespace", namespace);
         }
         deploymentNamespaces.clear();
-        LOGGER.info("Using namespace {}", CLUSTER.defaultNamespace());
-        setNamespace(CLUSTER.defaultNamespace());
+        LOGGER.info("Using namespace {}", defaultNamespace);
+        setNamespace(defaultNamespace);
     }
 
     /**

--- a/test/src/main/java/io/strimzi/test/k8s/KubeCluster.java
+++ b/test/src/main/java/io/strimzi/test/k8s/KubeCluster.java
@@ -38,9 +38,9 @@ public interface KubeCluster {
      * Returns the cluster named by the TEST_CLUSTER environment variable, if set, otherwise finds a cluster that's
      * both installed and running.
      * @return The cluster.
-     * @throws RuntimeException If no running cluster was found.
+     * @throws NoClusterException If no running cluster was found.
      */
-    static KubeCluster bootstrap() {
+    static KubeCluster bootstrap() throws NoClusterException {
         Logger logger = LogManager.getLogger(KubeCluster.class);
 
         KubeCluster[] clusters = null;
@@ -79,7 +79,8 @@ public interface KubeCluster {
             }
         }
         if (cluster == null) {
-            throw new RuntimeException("Unable to find a cluster; tried " + Arrays.toString(clusters));
+            throw new NoClusterException(
+                    "Unable to find a cluster; tried " + Arrays.toString(clusters));
         }
         return cluster;
     }

--- a/test/src/main/java/io/strimzi/test/k8s/KubeClusterResource.java
+++ b/test/src/main/java/io/strimzi/test/k8s/KubeClusterResource.java
@@ -4,6 +4,8 @@
  */
 package io.strimzi.test.k8s;
 
+import org.junit.jupiter.api.Assumptions;
+
 /**
  * A Junit resource which discovers the running cluster and provides an appropriate KubeClient for it,
  * for use with {@code @BeforeAll} (or {@code BeforeEach}.
@@ -76,7 +78,11 @@ public class KubeClusterResource {
 
     public KubeCluster cluster() {
         if (cluster == null) {
-            this.cluster = KubeCluster.bootstrap();
+            try {
+                this.cluster = KubeCluster.bootstrap();
+            } catch (NoClusterException e) {
+                Assumptions.assumeTrue(false, e.getMessage());
+            }
         }
         return cluster;
     }
@@ -84,7 +90,11 @@ public class KubeClusterResource {
     public void before() {
         if (bootstrap) {
             if (cluster == null) {
-                this.cluster = KubeCluster.bootstrap();
+                try {
+                    this.cluster = KubeCluster.bootstrap();
+                } catch (NoClusterException e) {
+                    Assumptions.assumeTrue(false, e.getMessage());
+                }
             }
             if (cmdClient == null) {
                 this.cmdClient = cluster.defaultCmdClient();

--- a/test/src/main/java/io/strimzi/test/k8s/Minikube.java
+++ b/test/src/main/java/io/strimzi/test/k8s/Minikube.java
@@ -45,6 +45,7 @@ public class Minikube implements KubeCluster {
         return new Kubectl();
     }
 
+    @Override
     public KubeClient defaultClient() {
         return new KubeClient(new DefaultKubernetesClient(CONFIG), "default");
     }

--- a/test/src/main/java/io/strimzi/test/k8s/Minishift.java
+++ b/test/src/main/java/io/strimzi/test/k8s/Minishift.java
@@ -44,6 +44,7 @@ public class Minishift implements KubeCluster {
         return new Oc();
     }
 
+    @Override
     public KubeClient defaultClient() {
         return new KubeClient(new DefaultOpenShiftClient(CONFIG), "myproject");
     }

--- a/test/src/main/java/io/strimzi/test/k8s/NoClusterException.java
+++ b/test/src/main/java/io/strimzi/test/k8s/NoClusterException.java
@@ -1,0 +1,11 @@
+/*
+ * Copyright 2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.test.k8s;
+
+public class NoClusterException extends Exception {
+    public NoClusterException(String message) {
+        super(message);
+    }
+}

--- a/test/src/main/java/io/strimzi/test/k8s/OpenShift.java
+++ b/test/src/main/java/io/strimzi/test/k8s/OpenShift.java
@@ -65,6 +65,7 @@ public class OpenShift implements KubeCluster {
         return new Oc();
     }
 
+    @Override
     public KubeClient defaultClient() {
         return new KubeClient(new DefaultOpenShiftClient(CONFIG), "myproject");
     }

--- a/topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorIT.java
+++ b/topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorIT.java
@@ -24,6 +24,8 @@ import io.strimzi.api.kafka.model.KafkaTopicBuilder;
 import io.strimzi.operator.common.Util;
 import io.strimzi.test.BaseITST;
 import io.strimzi.test.TestUtils;
+import io.strimzi.test.k8s.KubeCluster;
+import io.strimzi.test.k8s.NoClusterException;
 import io.vertx.core.Vertx;
 import io.vertx.ext.unit.Async;
 import io.vertx.ext.unit.TestContext;
@@ -45,6 +47,7 @@ import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.junit.After;
 import org.junit.AfterClass;
+import org.junit.Assume;
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Ignore;
@@ -104,6 +107,11 @@ public class TopicOperatorIT extends BaseITST {
 
     @BeforeClass
     public static void setupKubeCluster() {
+        try {
+            KubeCluster.bootstrap();
+        } catch (NoClusterException e) {
+            Assume.assumeTrue(e.getMessage(), false);
+        }
         cmdKubeClient()
                 .createNamespace(NAMESPACE);
         oldNamespace = setNamespace(NAMESPACE);
@@ -115,18 +123,20 @@ public class TopicOperatorIT extends BaseITST {
 
     @AfterClass
     public static void teardownKubeCluster() {
-        cmdKubeClient()
-                .delete("src/test/resources/TopicOperatorIT-rbac.yaml")
-                .delete(TestUtils.CRD_TOPIC)
-                .delete("../install/topic-operator/02-Role-strimzi-topic-operator.yaml")
-                .deleteNamespace(NAMESPACE);
-        cmdKubeClient().namespace(oldNamespace);
+        if (oldNamespace != null) {
+            cmdKubeClient()
+                    .delete("src/test/resources/TopicOperatorIT-rbac.yaml")
+                    .delete(TestUtils.CRD_TOPIC)
+                    .delete("../install/topic-operator/02-Role-strimzi-topic-operator.yaml")
+                    .deleteNamespace(NAMESPACE);
+            cmdKubeClient().namespace(oldNamespace);
+        }
     }
 
     @Before
     public void setup(TestContext context) throws Exception {
         LOGGER.info("Setting up test");
-        CLUSTER.before();
+        kubeCluster().before();
         Runtime.getRuntime().addShutdownHook(kafkaHook);
         int counts = 3;
         do {

